File: ServerStatusNotifierModel/src/main/java/com/github/grzegorzekkk/serverstatusnotifier/serverstatusnotifiermodel/SrvConnDetails.java
package com.github.grzegorzekkk.serverstatusnotifier.serverstatusnotifiermodel;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SrvConnDetails implements Serializable {
    private String address;
    private int port;
    private String password;
}

File: ServerStatusNotifier/app/src/main/java/com/github/grzegorzekkk/serverstatusnotifier/database/SrvConnDetails.kt
package com.github.grzegorzekkk.serverstatusnotifier.database

import android.os.Parcel
import android.os.Parcelable

data class SrvConnDetails(val address: String, val port: Int, val password: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readInt(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(address)
        parcel.writeInt(port)
        parcel.writeString(password)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SrvConnDetails> {
        override fun createFromParcel(parcel: Parcel): SrvConnDetails {
            return SrvConnDetails(parcel)
        }

        override fun newArray(size: Int): Array<SrvConnDetails?> {
            return arrayOfNulls(size)
        }
    }

}